backend/src/main/resources/application.properties

```diff
@@ -1,20 +1,23 @@
-server.port=8080
-
 spring.profiles.active=dev
 
+# REST related config
+server.servlet.context-path = /api/v1
+
+server.port=8080
+
 spring.data.neo4j.uri=bolt://localhost:7687
-spring.data.neo4j.username=neo4j
-spring.data.neo4j.password=stonecap
+spring.data.neo4j.username=${NEO4J_USERNAME}
+spring.data.neo4j.password=${NEO4J_PASSWORD}
 
 # Alpaca Related properties
 # Client ID Auth: 198903d0d2f523a25e4dce65837bbf0d
 # Client Secret Auth: d756cb7ae52ad380d16aec004a38b00f04747cf2
 alpaca.api.version=v2
-alpaca.key.id=PKZJK3IXQ4JZAWCBDSRX
-alpaca.secret=SA5sYIYR58AukN8NKpdGkQZSCvOCoTXYgyjuaoUe
+alpaca.key.id=${ALPACA_KEY_ID}
```

Comment:
You cannot use the secrets defined on our repo from here. You have to inject them from the CI/CD workflow. I believe there may be a way to achieve this but I haven't looked into it yet for Spring.

Reply:
The values are injected if I have them present within the environment variable of my computer. I tested it locally it works well

Even in the tests

Reply:
Maybe I misunderstood your comment?

Reply:
So GitHub Actions secrets can only be used within the DevOps pipeline workflow. It doesn't see the secrets at all on your local machine. It has to go through the CI/CD to be able to use the secret token. So I think the best solution here would be to separate everything into 3 environments (local, CICD and prod).
- If we test the endpoints locally and we need to use some API keys, we could use the sandbox keys. (need to figure out a way to hide the key though)
- If we test on the CICD, then we can use the secrets easily with 
```
${{ secrets.TOKEN_NAME }}
```
- If we deploy, we need to add all needed keys in the classpath of our JAR file before packaging it in the CD. Again, it can be easily achieved with something like 
```
jar -cp ALPACA_KEY=${{ secrets.ALPACA }} etc...
``` 

With those 3 cases, we need to detect which env we are in and apply whatever case above.

Reply:
I think there is some confusion here so I'll try to clarify. What we are trying to achieve here is hide the keys / make the application packaging independent of the keys. In order to do so, we make use of environment variables in the different environments where the application will run. The application is going to pick up these variables when running. The only reason we also store them in github secrets, is to be able to expose them as environment variables in the ci environment without exposing them in the ci.yml file.